tests/misc/print_format/testcase.yaml

```
@@ -0,0 +1,36 @@
+common:
+  harness: console
+  harness_config:
+    type: multi_line
+    ordered: true
+    regex:
+    - "d8"
+    - "d16"
+    - "d32"
+    - "d64"
+    - "i8"
+    - "i16"
+    - "i32"
+    - "i64"
+    - "o10"
+    - "o20"
+    - "o40"
+    - "o100"
+    - "u8"
+    - "u16"
+    - "u32"
+    - "u64"
+    - "x8"
+    - "x10"
+    - "x20"
+    - "x40"
+    - "X8"
+    - "X10"
+    - "X20"
+    - "X40"
+
+tests:
+  printk.format: {}
+  printk.format_newlib:
```

same here, also should filter out non-newlib toolchains

```suggestion
  printk.format_newlib:
    tags: clib newlib
    filter: TOOLCHAIN_HAS_NEWLIB == 1
```

---

Done